pkg/pgmodel/ingestor/trace/trace_batcher.go

```diff
@@ -0,0 +1,214 @@
+package trace
+
+import (
+	"context"
+	"runtime"
+	"sync"
+	"time"
+
+	"github.com/timescale/promscale/pkg/log"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+const (
+	defaultReqBufferSize      = 10000                  // tweaking this does not result in significant perf impact
+	defaultBatchSize          = 5000                   // this is >= (we might produce bigger batches sometimes)
+	defaultBatchTimeout       = 200 * time.Millisecond // arbitrary picked. We don't want client to wait too long
+	defaultMaxBufferedBatches = 200                    // arbitrary picked. We want to avoid wait on batches
+)
+
+var defaultBatchWorkers = runtime.NumCPU() / 2 // we only take half so other half can be used for writers
```

Comment:
This would cause a trouble when the promscale is configured with cpu limit of < 1.

Reply:
Right. Great catch. Will fix.